Sources/Sentry/Profiling/SentryProfilerSerialization.mm

```diff
@@ -196,9 +207,138 @@
     return payload;
 }
 
+NSMutableDictionary<NSString *, id> *
+sentry_serializedContinuousProfileChunk(
+    SentryId *profileID, NSDictionary<NSString *, id> *profileData, uint64_t startSystemTime,
+    uint64_t endSystemTime, NSDictionary<NSString *, id> *serializedMetrics,
+    NSArray<SentryDebugMeta *> *debugMeta, SentryHub *hub, uint64_t continuousChunkStartSystemTime
+#    if SENTRY_HAS_UIKIT
+    ,
+    SentryScreenFrames *gpuData
+#    endif // SENTRY_HAS_UIKIT
+)
+{
+    NSMutableArray<SentrySample *> *const samples = profileData[@"profile"][@"samples"];
+    // !!!: assumption: in legacy, we would avoid sending a payload with less than 2 samples. now,
+    // we may have previously sent a chunk with many samples, and then this chunk may be the last of
+    // the continuous profiling session and it only has 1 sample. assuming we'll want to keep that
+    // sample. let the backend decide, or, keep track of some statistics of the profiling session,
+    // to count the number of total samples sent so far, to decide whether or not a 1-sample chunk
+    // is acceptable.
+
+    const auto payload = [NSMutableDictionary<NSString *, id> dictionary];
+    NSMutableDictionary<NSString *, id> *const profile = [profileData[@"profile"] mutableCopy];
+    profile[@"samples"] = _sentry_serializedSamplesWithRelativeTimestamps(
+        samples, startSystemTime, SentryProfilerModeContinuous);
+
+    payload[@"profile"] = profile;
+
+    payload[@"version"] = @"2";
+    const auto debugImages = [NSMutableArray<NSDictionary<NSString *, id> *> new];
+    for (SentryDebugMeta *debugImage in debugMeta) {
+        [debugImages addObject:[debugImage serialize]];
+    }
```

Comment:
`m`: We could significantly reduce the payload if we only add the debug images references in the samples payload; see `SentryDebugImageProvider.getDebugImagesForFrames`, but we don't have to do this in this PR.

Reply:
This is good feedback. Maybe we send the full list of debug images in the first chunk, and then in subsequent chunks, send a list of images that have been added or removed since the last chunk. @phacops what do you think?

In any case I think this will need to be done in a subsequent PR, and any change is going to depend on a change in the backend as well, so we shouldn't hold up this PR for it. I've made a note in the [planning doc](https://www.notion.so/sentry/Continuous-Profiling-Cocoa-SDK-9d64a809ed2d487b89e4aa0358bfac18?pvs=4#c2725f3879c442729e8bd107dd56ea8c).

Reply:
Oh yes definitely after another app launch. I was only thinking about doing it between chunk envelopes in the same profile session (another app launch implies a different profile session)